### PR TITLE
Update Reliability and Availability document and AMI/Installer layout

### PR DESCRIPTION
**Pull request by KFishner**
There are a few changes in this PR:

- Update Reliability and Availability document to be more clear about the differences in backup/restore strategy for each operational mode. [1]
- Change the left navigation to separate out AMI and Installer docs, as we have more installer-specific materials now https://www.dropbox.com/s/r0im71a5fz0uru7/Screenshot%202018-03-27%2012.38.18.png?dl=0
- Move the "Upgrades" section that was part of the Installer install instructions to its own page

[1]
download

### CODE REVIEW

File: content/source/docs/enterprise/private/reliability-availability.html.md [hunk 1]
```diff
@@ -119,97 +119,58 @@ stores there.
...
+### Mounted Disk
 
-The frequency of automated snapshots can be configured such that worst-case
-data loss can be as low as 1 hour.
+_PostgreSQL Database_ and _Blob Storage_ use mounted disks for their
+data. Backup and restore of those volumes is the responsibility of the user, the
+system does not manage that.
 
-#### Mounted Disk
+_Configuration Data_ and _Vault Data_ for the installation is stored in Docker
```
Comment (KFishner) [inline content/source/docs/enterprise/private/reliability-availability.html.md]: @evanphx wanted to validate this with you - is it accurate that the Vault data is stored in Docker volumes

File: content/source/docs/enterprise/private/reliability-availability.html.md [hunk 2]
```diff
@@ -226,30 +187,48 @@ The procedure here would be as follows:
...
 along with the software.
 
-#### External Services
+### External Services
+
+In the External Services mode of the Installer Architecture, the
+**Application Layer** and **Coordination Layer** execute on a Linux instance,
+but the **Storage Layer** is configured to use external services in the form an
+a PostgreSQL server, an S3-compatible Blob Storage, and an
+[external Vault](./vault.html).
```
Comment (KFishner) [inline content/source/docs/enterprise/private/reliability-availability.html.md]: @evanphx flagging this change (referencing the external Vault guide) for you to review specifically

File: content/source/docs/enterprise/private/reliability-availability.html.md [hunk 3]
```diff
@@ -119,97 +119,58 @@ stores there.
 ~> **Note**: The PTFE Installer architecture is currently in beta and is being
 actively developed. The section below is subject to change as work continues.
 
-The PTFE Installer architecture has several supported modes of operation, each
-of which has different implications for reliability and availability.
-
-The operational mode is selected at install time and can not be changed
+This section describes how to setup your PTFE deployment to recover from
```
Comment (armchairlinguist) [inline content/source/docs/enterprise/private/reliability-availability.html.md]: setup > set up

File: content/source/docs/enterprise/private/reliability-availability.html.md [hunk 4]
```diff
@@ -119,97 +119,58 @@ stores there.
...
-available to be configured and automated from the Admin Console.
-
-### Mounted Volumes
-
-In the Mounted Volumes mode of the Installer Architecture, the
-**Application Layer**, **Coordination Layer**, and **Storage Layer** execute on
-a Linux instance.
+The below table explains where each data type in the Storage Layer is stored and
+the corresponding snapshot and restore procedure. For the data types that use
+PTFE's built-in snapshot and restore function managed by PTFE, follow
```
Comment (armchairlinguist) [inline content/source/docs/enterprise/private/reliability-availability.html.md]: > PTFE's built-in snapshot and restore function managed by PTFE
seems like probably both of these PTFE bits aren't needed?

File: content/source/docs/enterprise/private/reliability-availability.html.md [hunk 5]
```diff
@@ -262,19 +241,37 @@ The procedure here would be as follows:
...
 when the Terraform Enterprise software is updated so that
 any restored instance has the newest version. This is important
 because the data stored in the external services is versioned along with the
 software.
+
+### Availability During Upgrades
+
+Upgrades for the Installer Architecture utilize the Installer Admin Console.
+Once an upgrade has been been detected (either online or airgap), the new code
+is imported and once ready, all services on the instance are restarted running
```
Comment (armchairlinguist) [inline content/source/docs/enterprise/private/reliability-availability.html.md]: I might make this two sentences, replacing "and" with a period.

File: content/source/docs/enterprise/private/reliability-availability.html.md [hunk 6]
```diff
@@ -262,19 +241,37 @@ The procedure here would be as follows:
...
+the new code. The expected downtime is between 30 seconds and 5 minutes,
+depending on if database updates have to be applied.
```
Comment (armchairlinguist) [inline content/source/docs/enterprise/private/reliability-availability.html.md]: perhaps "depending on whether"?

File: content/source/docs/enterprise/private/reliability-availability.html.md [hunk 7]
```diff
@@ -119,97 +119,58 @@ stores there.
...
+volume storing the data is not lost. Because only Configuration and Vault data is stored
 on the instance, we recommend using a system snapshot mechanism to provide fast
-recovery.
+recovery following these steps:
 
-The procedure here would be as follows:
+**Snapshot**
 
 - Be sure that the mounted volume being used is automatically mounted
   automatically on start.  For instance with AWS, this means having it attached
```
Comment (blalor) [inline content/source/docs/enterprise/private/reliability-availability.html.md]: "automatically mounted automatically" is redundant.

Comment (blalor) [inline content/source/docs/enterprise/private/reliability-availability.html.md]: Add `,` between "For instance" and "with AWS".

File: content/source/docs/enterprise/private/reliability-availability.html.md [hunk 8]
```diff
@@ -226,30 +187,48 @@ The procedure here would be as follows:
   instance. The exact mechanism used is specific to the virtualization
   environment that the product runs in.
 
+**Restore**
+
 - Configure the virtualization to run the new snapshot and restart a new
   instance should it stop. In AWS, this is done using an Autoscaling Group.
   The environment can monitor the availability of port 443 for instance health
   as well.
```
Comment (blalor) [inline content/source/docs/enterprise/private/reliability-availability.html.md]: This is inaccurate.  The Replicated config contains references to the instance's IP address, which will change (even if an instance is stopped and then started).

File: content/source/docs/enterprise/private/reliability-availability.html.md [hunk 9]
```diff
@@ -262,19 +241,37 @@ The procedure here would be as follows:
   instance. The exact mechanism used is specific to the virtualization
```
Comment (blalor) [inline content/source/docs/enterprise/private/reliability-availability.html.md]: My previous comment on snapshotting the instance applies here, as well.

File: content/source/docs/enterprise/private/installer.html.md [hunk 10]
```diff
@@ -0,0 +1,16 @@
+---
+layout: "enterprise2"
+page_title: "Private Terraform Enterprise Installer"
+sidebar_current: "docs-enterprise2-private-installer"
+---
+
+# Private Terraform Enterprise
+
+This section is guides users through deploying Private Terraform Enterprise
```
Comment (nfagerlund) [inline content/source/docs/enterprise/private/installer.html.md]: typo: "is guides"

File: content/source/docs/enterprise/private/reliability-availability.html.md (hunk 1 above)
Comment (evanphx) [reply content/source/docs/enterprise/private/reliability-availability.html.md]: No, it's not correct. The vault key material is stored on the host in `/var/lib/tfe-vault` and the actual data that vault uses is in Postgres.

File: content/source/docs/enterprise/private/reliability-availability.html.md (hunk 2 above)
Comment (evanphx) [reply content/source/docs/enterprise/private/reliability-availability.html.md]: That's fine but it should be spelled out that using an external Vault is an optional configuration, not something that is used by default with external services mode.

File: content/source/docs/enterprise/private/reliability-availability.html.md (hunk 7 above)
Comment (KFishner) [reply content/source/docs/enterprise/private/reliability-availability.html.md]: I'm going to remove the `,` on this actually! You only add a comma if there are three prepositional phrases at the start of the sentence

`For example with AWS in the us-east region, you need to add a comma`

**Comment by FraserPol** [comment]
👍

**Comment by phinze** [comment]
@KFishner I took a stab at resolving the conflicts here but I got stuck - think we might need to start from a fresh copy and re-do the edits you were going for based on the latest content.

**Comment by blalor** [comment]
Going back through this guide, I think it would be simpler to just reference the automated recovery page for the External Services and Mounted Disk configurations.  The specifics of that configuration are more clearly called out there.  As it's written right now, I feel like the "snapshot" and "restore" sections are a mix of specific ("from the admin UI, click *Stop*") and general ("create an image of the stopped instance").

As I commented previously, snapshotting the instance is unfortunately not sufficient, as Replicated bakes details about the running instance into config files, and those details are no longer relevant when booting a snapshotted instance on different (virtual) hardware.  I think the spirit of the guide can be preserved while simply removing the Snapshot and Restore sections from those two configuration overviews.

**Comment by KFishner** [comment]
@blalor good call, I agree! 

@phinze merge conflicts should be good 👍 merge into master when ready

**Pull request merged**